modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java

```diff
@@ -549,7 +553,7 @@ private static boolean eq(Object left, Object right) {
                 if (left instanceof Number) {
                     return ((Number) left).floatValue() == (float) right;
                 } else if (left instanceof Character) {
-                    return (char) left == ((Number) right).floatValue();
+                    return (char) left == (float) right;
```

Comment:
This is not covered by a test, can you add some coverage?

Reply:
Added